src/effects/native/phasereffect.cpp

```diff
@@ -0,0 +1,170 @@
+    fb->setMaximum(1.0);
+
+    EffectManifestParameter* sweep = manifest.addParameter();
+    sweep->setId("sweep_width");
+    sweep->setName(QObject::tr("Sweep"));
+    sweep->setDescription("Sets sweep width.");
+    sweep->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
+    sweep->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    sweep->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    sweep->setDefault(0.5);
+    sweep->setMinimum(0.0);
+    sweep->setMaximum(1.0);
+
+    return manifest;
+}
+
+PhaserEffect::PhaserEffect(EngineEffect* pEffect,
+                           const EffectManifest& manifest) 
+        : m_pStagesParameter(pEffect->getParameterById("stages")),
+          m_pFrequencyParameter(pEffect->getParameterById("frequency")),
+          m_pDepthParameter(pEffect->getParameterById("depth")),
+          m_pFeedback(pEffect->getParameterById("feedback")),
+          m_pSweepWidth(pEffect->getParameterById("sweep_width")) {
+    Q_UNUSED(manifest);
+}
+
+PhaserEffect::~PhaserEffect() {
+    //qDebug() << debugString() << "destroyed";
+}
+
+void PhaserEffect::processChannel(const ChannelHandle& handle,
+                                  PhaserGroupState* pState,
+                                  const CSAMPLE* pInput, CSAMPLE* pOutput,
+                                  const unsigned int numSamples,
+                                  const unsigned int sampleRate,
+                                  const EffectProcessor::EnableState enableState,
+                                  const GroupFeatureState& groupFeatures) {
+
+    Q_UNUSED(handle);
+    Q_UNUSED(enableState);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(sampleRate);
+
+    CSAMPLE frequency = m_pFrequencyParameter->value();
+    CSAMPLE depth = m_pDepthParameter->value();
+    CSAMPLE feedback = m_pFeedback->value();
+    CSAMPLE sweepWidth = m_pSweepWidth->value();
+    int stages = m_pStagesParameter->value();
+
+    CSAMPLE* filterLeft = pState->filterCoefLeft;
+    CSAMPLE* filterRight = pState->filterCoefRight;
+    CSAMPLE* oldInLeft = pState->oldInLeft;
+    CSAMPLE* oldOutLeft = pState->oldOutLeft;
+    CSAMPLE* oldInRight = pState->oldInRight;
+    CSAMPLE* oldOutRight = pState->oldOutRight;
+
+    CSAMPLE left = 0, right = 0;
+    CSAMPLE leftPhase; 
+    CSAMPLE rightPhase;
+    CSAMPLE freqSkip = frequency * 2.0 * M_PI / sampleRate;
+
+    int counter = 0;
+    int updateCoef = 32;
+
+    const int kChannels = 2;
+    for (unsigned int i = 0; i < numSamples; i += kChannels) {
+
+        pState->time++;
+        left = pInput[i] + left * feedback; 
+        right = pInput[i + 1] + right * feedback;
+
+        leftPhase = fmodf(freqSkip * pState->time, 2.0 * M_PI);
+        rightPhase = fmodf(freqSkip * pState->time + M_PI, 2.0 * M_PI);
+
+        if ((counter++) % updateCoef == 0) {
+            for (int j = 0; j < stages; j++) {
+                CSAMPLE delayLeft = 0.5 + 0.5 * sin(leftPhase);
```

Comment:
This part of the code is confusing. What is delayLeft, really a sample value, or a time? 
It changes the content though the function. It would be nice to use a new speaking name every time the content changes. (The compiler will throw them out anyway)